### PR TITLE
Update vNext version in azure-dev.yaml

### PR DESCRIPTION
**Pull request by mboersma**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Updates the psuedo-version used to point to the "next" CAPZ release.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/config/azure-dev.yaml [hunk 1]
```diff
@@ -92,7 +92,7 @@ providers:
       replacements:
       - old: "imagePullPolicy: Always"
         new: "imagePullPolicy: IfNotPresent"
-    - name: v1.9.99 # "vNext"; use manifests from local source files
+    - name: v1.10.99 # "vNext"; use manifests from local source files
```
Comment (CecileRobertMichon) [inline test/e2e/config/azure-dev.yaml:95]: we also need to update the test metadata as described in https://capz.sigs.k8s.io/developers/releasing.html#update-test-capz-provider-metadatayaml-skip-for-patch-releases

**Review by jackfrancis — APPROVED** [review]
/lgtm
/approve

File: metadata.yaml [hunk 2]
```diff
@@ -44,3 +44,6 @@ releaseSeries:
   - major: 1
     minor: 9
     contract: v1beta1
+  - major: 1
```
Comment (CecileRobertMichon) [inline metadata.yaml:47]: this PR should update https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/test/e2e/data/shared/v1beta1_provider/metadata.yaml (this one is already updated in the main branch)

Comment (mboersma) [reply metadata.yaml]: 🤦🏻 updated!

**Comment by jackfrancis** [comment]
thanks @CecileRobertMichon for finding that in time!

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm

**Pull request merged**